rule/package-comments.go

```diff
@@ -31,9 +31,48 @@ func (*PackageCommentsRule) Apply(file *lint.File, _ lint.Arguments) []lint.Fail
 	fileAst := file.AST
 	w := &lintPackageComments{fileAst, file, onFailure}
 	ast.Walk(w, fileAst)
+	failures = append(failures, checkPackageComment(file.Pkg)...)
```

Comment:
the previous line is the original "walk" (not a real one in fact) of the file AST to check for failures. How that analysis relates with the one performed by ` checkPackageComment` ? They are complementary? Or the new check totally replaces the original one?

Reply:
maybe we should do checking in `Visit`